File: frontend/app/services/notification-service.ts
// services/notification-service.ts

import { apiClient } from "../api/api-client";


export interface NotificationItem {
  id: number;
  user_id: number;
  content: string;
  is_read: boolean;
  created_at: string;
  post_id: number | null;
  sender_id: number | null;
  type: string;
}




export const NotificationService = {
  async getAllNotifications(userId: number) {
    return apiClient.get<{ status: string; message: string; data: NotificationItem[] }>(
      `/notifications/${userId}`
    );
  },

  async markNotificationAsRead(notificationId: number) {
    return apiClient.put<{ status: string; message: string }>(
      `/notifications/read/${notificationId}`
    );
  },

  async deleteNotification(notificationId: number) {
    return apiClient.delete<{ status: string; message: string }>(
      `/notifications/delete/${notificationId}`
    );
  },

  async deleteAllNotifications(userId: number) {
    return apiClient.delete<{ status: string; message: string }>(
      `/notifications/delete/all/${userId}`
    );
  },

  async markAllNotificationsAsRead(userId: number) {
    return apiClient.put<{ status: string; message: string }>(
      `/notifications/read/all/${userId}`
    );
  },
};

File: mobile/lib/src/features/notification/data/notification_service.dart
import 'package:dio/dio.dart';
import 'package:seattle_pulse_mobile/src/core/api/api_client.dart';
import 'package:seattle_pulse_mobile/src/features/notification/data/notification_model.dart';

class NotificationService {
  final ApiClient _client = ApiClient();

  Future<List<NotificationItem>> getAllNotifications(int userId) async {
    Response response = await _client.get('/notifications/$userId');
    final data = response.data['data'] as List;
    print("data $data");
    return data.map((json) => NotificationItem.fromJson(json)).toList();
  }

  Future<void> markNotificationAsRead(int notificationId) async {
    await _client.put('/notifications/read/$notificationId');
  }

  Future<void> markAllNotificationsAsRead(int userId) async {
    await _client.put('/notifications/read/all/$userId');
  }

  Future<void> deleteNotification(int notificationId) async {
    await _client.delete('/notifications/delete/$notificationId');
  }

  Future<void> deleteAllNotifications(int userId) async {
    await _client.delete('/notifications/delete/all/$userId');
  }
}
